### PR TITLE
Add regression test for DAE and separate compile

### PR DESCRIPTION
**Pull request by AlexeySachkov**

### CODE REVIEW

File: SYCL/Regression/DAE-separate-compile.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  sycl::buffer buf2(&storage2, sycl::range<1>(1));
+
+  q.submit([&](sycl::handler &h) {
+    auto acc = buf.get_access(h);
+    auto acc2 = buf2.get_access(h);
+    h.single_task([=]() {
+      (void)acc; // unused
+      acc2[0] = THE_ANSWER;
+    });
+  });
```
Comment (steffenlarsen) [inline SYCL/Regression/DAE-separate-compile.cpp]:
```suggestion
  int storage2;
  {
    sycl::buffer buf(&storage, sycl::range<1>(1));
    sycl::buffer buf2(&storage2, sycl::range<1>(1));

    q.submit([&](sycl::handler &h) {
      auto acc = buf.get_access(h);
      auto acc2 = buf2.get_access(h);
      h.single_task([=]() {
        (void)acc; // unused
        acc2[0] = THE_ANSWER;
      });
    });
  }
```
This should ensure that the data from `buf2` has been written back to `storage2` before the check.

File: SYCL/Regression/DAE-separate-compile.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// There was an issue with dead argument elminitaion optimization, which lead
+// to runtime errors when setting kernel arguments if the app was compiled with
+// optimizations, but linked without them.
+//
+// The test checks that the scenario works correctly.
+//
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -O2 -c -o %t.o
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -O0 -o %t.out
```
Comment (AlexeySachkov) [inline SYCL/Regression/DAE-separate-compile.cpp]:
```suggestion
// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %t.o -O0 -o %t.out
```

File: SYCL/Regression/DAE-separate-compile.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+// The test checks that the scenario works correctly.
+//
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -O2 -c -o %t.o
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %t.o -O0 -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <iostream>
+#include <CL/sycl.hpp>
```
Comment (AlexeySachkov) [inline SYCL/Regression/DAE-separate-compile.cpp]:
```suggestion
#include <sycl/sycl.hpp>
#include <iostream>
```

File: SYCL/Regression/DAE-separate-compile.cpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+// The test checks that the scenario works correctly.
+//
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -O2 -c -o %t.o
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %t.o -O0 -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <sycl/sycl.hpp>
+#include <iostream>
```
Comment (AlexeySachkov) [inline SYCL/Regression/DAE-separate-compile.cpp]:
```suggestion
#include <iostream>
#include <sycl/sycl.hpp>
```

**Review by steffenlarsen — APPROVED** [review]

**Pull request merged**